### DISCUSSION TITLE
Tuya TX/RX vs Other components?

### DISCUSSION THREAD

**Discussion by ericanastas** [arendst/Tasmota#10394] [Q&A] [answered]
I'm just getting started with HA, and currently do not have any smart devices. I've been doing some research and a Wifi based system using Tasmota sounds like the most flexible and future proof option. So now I'm looking into getting a few smart switches to get started. I'll probably just start off using Tuya, but want something I can flash to Tasmota in the future.

Can someone explain the differences between the GPIO components on the two models below

- [Tessan MJ-SD02 Dimmer
](https://templates.blakadder.com/tessan_SD-02.html)
    This model lists separate components for Button 1, Button 2, Button 3, Relay 1, PWM, LED 1, LED 2, LED 3.

- [TreatLife DS02S Dimmer
](https://templates.blakadder.com/treatlife_DS02S.html)
    While this model only has components for Tuya RX and Tuya TX?

My impression is the  Tessan model would be more flexible in the long run as it provides separate access to the buttons, relay and LEDs. Is this correct? What are Tuna TX/RX components?

**Comment by sfromis** [comment, marked as answer]
The Tuya Tx/Rx pins are when the device is controlled by anther chip, "TuyaMCU", with the ESP8266 only being used as a wifi bridge. You are right in your impression of it being more flexible to have pin access directly. Basically, the TuyaMCU setup restricts the device to same functions as controllable from the app, cutting out a lot of the additional features possible with Tasmota.
